shm/Map.cpp

```diff
@@ -0,0 +1,46 @@
+#include "Map.hpp"
+#include <algorithm>
+#include <random>
+#include <cstddef>
```

Comment:
Here you have header and standard library include in the same paragraph. And includes should be sorted alphabetically,so cstddef should be before random.